python/tank/util/shotgun_entity.py

```diff
@@ -331,47 +333,54 @@ def _generate_name(self, expression, values):
                     field_def["regex_obj"],
                 )
 
-            str_data[full_sg_field_name] = str_value
+            str_data[token] = str_value
 
-        # change format from {xxx} to %(xxx)s for value substitution.
-        # make sure we remove any expression tokens from folder:
-        # {xxx} -> %(xxx)s
-        # {xxx:yyy} -> %(xxx)s
-        adjusted_expr = re.sub("\{([^:\}]+)[^\}]*\}", "%(\\1)s", expression)
+        # Now str_data looks something like
+        # {'code': 'hello', 'code:^(.)': 'h'}.
```

Comment:
Just out of curiosity, what is this syntax `code:^(.)`?

Reply:
Ah, I think I get it now. It tells Toolkit to match ^(.) in the name and returns what `(.)`, captured, right? Didn't know this was a feature!

How often is this used?

Reply:
oh it's brand new - part of this feature (#46177) - basically extends the folder schema so that it works similar to the substring syntax that we have for templates!

Reply:
yep, so basically run the regex, extract the `(tokens)` (if there is more than one, concatenate them together) and that forms the value that is used to form the folder name on disk. Makes it easy to slice and dice things, extract different parts of a string etc.